mkosi/installer/__init__.py

```diff
@@ -41,6 +41,8 @@ def mounts(cls, context: Context) -> list[PathString]:
 
         for d in ("cache", "lib"):
             src = context.package_cache_dir / d / subdir
+            if not src.exists():
```

Comment:
I think `--bind-try` would be the thing to use for that case:
> Equal to `--bind` but ignores non-existant SRC

I vaguely remember @DaanDeMeyer mentioning, that CentOS8's bubblewrap was to old for that, but according to pkgs.org that has version 0.4 and I can find it in the man page for version 0.3.1, so it might be safe.

Reply:
`--ro-bind-try` is already in use, so I'll rework this to use `--bind-try`.

Reply:
It's specifically dnf assuming that the directory exists that causes the failure first. Other package managers only need it to be a virtual path and append a subdir to it.

The later part where it additionally binds the default package cache dir in checks whether the directory exists first, so shouldn't need `--bind-try`.


```diff
diff --git a/mkosi/installer/__init__.py b/mkosi/installer/__init__.py
index 3ee2f903..d86c17c8 100644
--- a/mkosi/installer/__init__.py
+++ b/mkosi/installer/__init__.py
@@ -41,9 +41,7 @@ class PackageManager:
 
         for d in ("cache", "lib"):
             src = context.package_cache_dir / d / subdir
-            if not src.exists():
-                continue
-            mounts += ["--bind", src, Path("/var") / d / subdir]
+            mounts += ["--bind-try", src, Path("/var") / d / subdir]
 
             # If we're not operating on the configured package cache directory, we're operating on a snapshot of the
             # repository metadata in the image root directory. To make sure any downloaded packages are still cached in
diff --git a/mkosi/installer/dnf.py b/mkosi/installer/dnf.py
index e74fd48c..0e5f0875 100644
--- a/mkosi/installer/dnf.py
+++ b/mkosi/installer/dnf.py
@@ -30,6 +30,8 @@ class Dnf(PackageManager):
 
     @classmethod
     def cache_subdirs(cls, cache: Path) -> list[Path]:
+        if not cache.exists():
+            return []
         return [
             p / "packages"
             for p in cache.iterdir()
```

Reply:
Sorry for the churn but I actually prefer the initial implementation in this case. The command lines are already long enough as they are no need to make them even longer.